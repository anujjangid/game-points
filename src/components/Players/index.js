import React from 'react';

import HeaderBar from '../HeaderBar';
import Blocks from '../Blocks';
import { Points, KahootTitle } from '../../Constants';

import { Column } from '../../styles';

const Players = () => {
  return (
    <Column span="8">
      {/* Header bar */}
      <HeaderBar title={KahootTitle} />
      {/* player blocks coming from constants */}
      <Blocks data={Points} />
    </Column>
  );
};

export default Players;
