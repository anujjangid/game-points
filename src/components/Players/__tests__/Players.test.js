import React from 'react';
import { create } from 'react-test-renderer';

import Players from '../index';

const props = {
  data: []
};

describe('Test suite for Players Component', () => {
  // create a UI snapshot to indentify UI changes or design regressions
  it('render correctly', () => {
    const component = create(<Players data={props} />);
    expect(component.toJSON()).toMatchSnapshot();
    const rootInstance = component.root;
    // check heading
    const Heading = rootInstance.findByType('h4');
    expect(Heading.props.children).toBe('Kahoot! POINTS');
    // finding div length
    const Block = rootInstance.findByType('section');
    expect(Block.props.children).toHaveLength(5);
  });
});
