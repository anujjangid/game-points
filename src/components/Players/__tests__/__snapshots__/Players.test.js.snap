// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Players Component render correctly 1`] = `
<div
  className="sc-EHOje ejagVo"
  span="8"
>
  <header
    className="sc-bdVaJa eQZKMo"
  >
    <h4>
      Kahoot! POINTS
    </h4>
  </header>
  <section
    className="sc-bwzfXH HjVAy"
  >
    <div
      className="sc-htpNat jJQQBl"
      onClick={[Function]}
    >
      A
    </div>
    <div
      className="sc-htpNat jJQQBl"
      onClick={[Function]}
    >
      B
    </div>
    <div
      className="sc-htpNat jJQQBl"
      onClick={[Function]}
    >
      C
    </div>
    <div
      className="sc-htpNat jJQQBl"
      onClick={[Function]}
    >
      D
    </div>
    <div
      className="sc-htpNat jJQQBl"
      onClick={[Function]}
    >
      E
    </div>
  </section>
</div>
`;
